src/frontend/src/session/cursor_manager.rs

```diff
@@ -525,9 +533,17 @@ impl CursorManager {
             handle_args,
         )
         .await?;
-        self.cursor_map
-            .lock()
-            .await
+        let mut cursor_map = self.cursor_map.lock().await;
+
+        cursor_map.retain(|_, v| {
+            if let Cursor::Subscription(cursor) = v {
+                !matches!(cursor.state, State::Invalid)
+            } else {
+                true
```

Comment:
I wonder whether we have similar issue for query cursor. I suspect that query cursor also suffer from the same issue of being valid after the table is dropped.